data/crac/crac-result-extensions/src/main/java/com/farao_community/farao/data/crac_result_extensions/JsonResultExtension.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.data.crac_result_extensions;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_api.Identifiable;
+import com.farao_community.farao.data.crac_impl.json.ExtensionsHandler;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.google.auto.service.AutoService;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * @author Baptiste Seguinot {@literal <baptiste.seguinot at rte-france.com>}
+ */
+
+@AutoService(ExtensionsHandler.ExtensionSerializer.class)
+public class JsonResultExtension<T extends Identifiable<T>>
+    implements ExtensionsHandler.ExtensionSerializer<T, ResultExtension<T, ? extends Result<T>>> {
+
+    @Override
+    public void serialize(ResultExtension<T, ? extends Result<T>> resultExtension, JsonGenerator jsonGenerator, SerializerProvider serializerProvider) throws IOException {
+        jsonGenerator.writeStartObject();
+
+        // serialize result per result as  the whole map somehow skip the result types
```

Comment:
// serialize result per result as the whole map somehow skips the result types

Reply:
better